orttraining/orttraining/training_api/include/onnxruntime_training_cxx_api.h

```diff
@@ -163,6 +163,24 @@ class TrainingSession : public detail::Base<OrtTrainingSession> {
                   const std::optional<std::basic_string<ORTCHAR_T>>& eval_model_path = std::nullopt,
                   const std::optional<std::basic_string<ORTCHAR_T>>& optimizer_model_path = std::nullopt);
 
+  /** \brief Create a training session that can be used to begin or resume training.
+   * This constructor allows the users to load the models from buffers instead of files.
+   *
+   * \param[in] env Env to be used for the training session.
+   * \param[in] session_options SessionOptions that the user can customize for this training session.
+   * \param[in] checkpoint_state Training states that the training session uses as a starting point for training.
+   * \param[in] train_model_data Buffer containing training model data.
+   * \param[in] train_model_data_length Length of the train_model_data buffer.
+   * \param[in] eval_model_data Buffer containing evaluation model data.
+   * \param[in] eval_model_data_length Length of the eval_model_data buffer.
+   * \param[in] optim_model_data Buffer containing optimizer model (used for performing weight/parameter update).
+   * \param[in] optim_model_data_length Length of the optim_model_data buffer.
+   *
+   */
+  TrainingSession(const Env& env, const SessionOptions& session_options, CheckpointState& checkpoint_state,
+                  const void* train_model_data, size_t train_model_data_length,
```

Comment:
I was actually hoping to use `std::span` but it seems like std::span can only be used in C++>=20. Not sure if we want to enforce users to use C++>20.

Shall we accept a `std::vector<uint8_t>` instead? How else might users pass in a raw buffer?